### PR TITLE
add materialized view support

### LINKED ISSUE #1021: Postgres materialized views problem
Hi!
Help needed.

SQLC 1.8.0
Database PostgreSQL 13.2

I have materialized view as shown below:

```
create materialized view m_lentities_assets as
    select l.id             as lentity_id,
           l.short_name     as lentity_short_name,
           l.full_name      as lentity_full_name,
           l.status         as lentity_status,
           m.role_id        as lentity_role_id,
           a.id             as asset_id,
           a.parent_id      as asset_parent_id,
           a.atypes_id      as asset_atypes_id,
           a.name           as asset_name,
           a.status         as asset_status
      from lentities l,
           assets a,
           lentity_assets m
     where m.lentity_id = l.id
       and m.asset_id = a.id
      with no data;

refresh materialized view m_lentities_assets;
```
and query respectively:

```
-- name: MLentitiesAssetsGet :many
select *
  from m_lentities_assets
 order by @sql_order::text
 limit @sql_limit offset @sql_offset;

```
but im unlucky yet...
Even for the read only selects i can't generate any result...

```
docker run --rm -v /Users/Stalker/workspace/src/gitlab.com/company/be:/src -w /src kjconroy/sqlc generate
# package dbs
storage/query/m_lentities_assets.sql:1:1: relation "m_lentities_assets" does not exist
make: *** [generate-dbs] Error 1

```
Any help would be appreciated.
Thanx!

### PR DESCRIPTION
**Pull request by ludusrusso**
This PR Adds materialized view support.

> Note that Materialized View are considered as `CreateTableAs` by `github.com/pganalyze/pg_query_go/v2` so this adds also supporto to `CREATE TABLE AS`

Fix #1021

@kylecarbs it was a little bit hard to understand that `CREATE MATERIALIZED VIEW` are treated  `CREATE TABLE AS` under the hood. It seems working, what do you think?

### CODE REVIEW

**Comment by kylecarbs** [comment]
@ludusrusso I _believe_ you intended on tagging @kyleconroy!

**Comment by ludusrusso** [comment]
😂😂 yep! Sorry!!

File: internal/sql/catalog/table.go [hunk 1]
```diff
@@ -247,3 +247,51 @@ func (c *Catalog) renameTable(stmt *ast.RenameTableStmt) error {
...
+		return sqlerr.RelationExists(tbl.Rel.Name)
+	}
+
+	schema.Tables = append(schema.Tables, &tbl)
+
+	// if stmt.Replace && err == nil {
+	// 	schema.Tables[existingIdx] = &tbl
+	// } else {
+	// 	schema.Tables = append(schema.Tables, &tbl)
+	// }
```
Comment (kyleconroy) [inline internal/sql/catalog/table.go:294]: No need to leave uncommented code around. Can you delete this?

File: internal/sql/catalog/catalog.go [hunk 2]
```diff
@@ -330,6 +331,9 @@ func (c *Catalog) Update(stmt ast.Statement, colGen columnGenerator) error {
 	case *ast.CreateTableStmt:
 		err = c.createTable(n)
 
+	case *ast.CreateTableAsStmt:
+		return c.createTableAs(n, colGen)
```
Comment (kyleconroy) [inline internal/sql/catalog/catalog.go:335]: Please match the current style in this function and set `err = c.createTableAs`

File: internal/sql/ast/matview_stmt.go [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+	Aliases         *List
+	Query           Node
+	Replace         bool
+	Options         *List
+	WithCheckOption ViewCheckOption
+}
+
+func (n *MatViewStmt) Pos() int {
+	return 0
+}
```
Comment (kyleconroy) [inline internal/sql/ast/matview_stmt.go:14]: I believe this code is unused and can be removed.

**Review by kyleconroy — CHANGES_REQUESTED** [review]
Since this is implemented via `CREATE TABLE AS`, can we also add a test case for that statement?

**Comment by ludusrusso** [comment]
> Since this is implemented via `CREATE TABLE AS`, can we also add a test case for that statement?

Fix your suggestions and add a new test case for `create table as`!
Thanks for help!

**Comment by malonehedges** [comment]
third party bump, looks like comments are addressed cc @kyleconroy 

was trying to use materialized views today so happy to see this on the finish line 😄

**Review by kyleconroy — APPROVED** [review]

**Pull request merged**